nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/shape/CreateView.java

```diff
@@ -74,6 +78,14 @@ public List<DataType> calculateOutputDataTypes(List<org.nd4j.linalg.api.buffer.D
         return Collections.singletonList(dataTypes.get(0));
     }
 
+    @Override
+    public List<SDVariable> doDiff(List<SDVariable> f1) {
+        List<SDVariable> ret = new ArrayList<>();
+        ret.add(f1.get(0));
+        ret.addAll(Arrays.asList(args()).subList(1, args().length));
+        return new CreateView(sameDiff,f1.get(0),Arrays.copyOfRange(args(),1,args().length)).outputs();
```

Comment:
why the ret variable here?  maybe I'm a bit blind, but I'm not seeing it being used for the returned value.

Reply:
Fixed. Was mid thinking there and missed that..